extensions/amp-tabs/0.1/amp-tabs.js

```diff
@@ -25,15 +26,79 @@ export class AmpTabs extends AMP.BaseElement {
   /** @param {!AmpElement} element */
   constructor(element) {
     super(element);
+
+    /** @private {?NodeList} */
+    this.sections_ = null;
   }
 
   /** @override */
   buildCallback() {
+    //TODO: handle case where there are > 1 <amp-tabs> on a page.
+
     user().assert(isExperimentOn(this.win, TAG),
         `Experiment "${TAG}" is disabled.`);
+
+    let selectedTabHeaderIndex = -1;
+    let selectedTabContentIndex = -1;
+    this.sections_ = this.getRealChildren();
+    this.sections_.forEach((section, index) => {
+      /** The first section should be the tab headers */
+      if (!index) {
```

Comment:
Let's use functionality from dom.js and array.js for these checks to simplify it a bit. I am thinking something along these lines (non-working code)

```
const tabsParent = childElementByTag(this.element, 'ul');
user().assert(tabsParent != null, 'Tab headers should be enclosed in a <ul> tag...');

const tabs = childElementsByTag(tabsParent, 'li');
user().assert(tabs.length > 0, 'There should be at least one tab items enclosed in a <li> tag...');

const sections = childElementsByTag(this.element, 'li');
user().assert(sections.length == tabs.length, 'There should same number of sections as tabs...');

const selectedTab = childElementByAttribute(tabs, 'selected' );
const selectedSection = childElementByAttribute(sections, 'selected' );

user().assert(selectedTab != null, 'You need to specify which..');
user().assert(selectedSection != null, 'You need to specify which..');
user().assert(tabs.indexOf(selectedTab) == sections.indexOf(selectedSection), 'Selected Tab Header and Selected Tab content indices do not match!');

this.sections_ = sections;
this.tabs_ = tabs;

tabs.forEach( (tab, index) => {
// Note that we are binding the tabs rather than an index
 tab.addEventListener('click', this.onHeaderClick_.bind(this, index));
});
```